packages/cli/src/workflow-runner.ts

```diff
@@ -286,12 +295,54 @@ export class WorkflowRunner {
 					data.executionData,
 				);
 				workflowExecution = workflowExecute.processRunExecutionData(workflow);
+			} else if (data.preferredTrigger?.data && data.startNodes && !data.destinationNode) {
+				this.logger.debug(
+					`Execution ID ${executionId} had preferredTrigger. Starting from that trigger.`,
+					{ executionId },
+				);
+				const startNodes = data.startNodes.map((data) => {
+					const node = workflow.getNode(data.name);
+					a.ok(node, `Could not find a node named "${data.name}" in the workflow.`);
```

Comment:
For learning - When do we use these assertions as opposed to actual errors?

Reply:
Interesting you ask, JP asked me yesterday as well 😄 

In my opinion they make sense for holding up invariants and get early feedback on programmer errors:
1. you checked before somewhere, like in the caller
		- that could be better done using more restrictive types, but this gets easily out of hand and turns academic
2. proving to typescript that something you know is true is actually true like putting something into a map and later in the same function taking it out, 
		- the assertion is still better than just slapping a `!` there in my opinion, just in case some refactor breaks this later, better to fail as early as possible
4. the FE sends correctly typed yet still invalid data, e.g. start node names that don't exist in the workflow
		- if this happens it's not a user error and they cannot fix it, the FE logic is flawed and it can't really deal with that error either, it's a bug and needs to be fixed

If the error is supposed to communicate with the user or FE and is actionable to either of them I'd create a proper error class.